tests/reftests/repository.test

```diff
@@ -607,3 +607,185 @@ opam-version: "2.0"
+[versions] synchronised from file://${BASEDIR}/VRS
+opam-file                       opam-version "5.0" unsupported on ${BASEDIR}/OPAM/repo/versions/packages/bad-opam-version/bad-opam-version.1/opam. Added as dummy unavailable package.
+Now run 'opam upgrade' to apply any package updates.
+### opam list -A --all-versions -s
+a.1
+bad-opam-version.1
+bad-opam-version.2
+foo.1
+foo.2
+foo.3
+foo.4
+foo.5
+foo.6
+good-opam-version.1
+### opam repository remove -a versions
+### opam repository remove -a repo
+### opam list -A --all-versions -s
+### opam repository --this-switch add repo ./REPO
+[repo] Initialised
+### opam repository --this-switch add versions ./VRS
+[versions] Initialised
+### opam list -A --all-versions -s
+a.1
+bad-opam-version.1
+bad-opam-version.2
+foo.1
+foo.2
+foo.3
+foo.4
+foo.5
+foo.6
+good-opam-version.1
+### opam show --raw bad-opam-version.1 | "${OPAMMAJORVERSION}" -> "CURRENTMAJOR"
+opam-version: "2.0"
+name: "bad-opam-version"
+version: "1"
+synopsis: ""
+description: """\
+This package uses opam 5.0 file format which opam CURRENTMAJOR cannot read.
+
+In order to install or view information on this package, please upgrade your opam installation to at least version 5.0."""
+available: opam-version >= "5.0"
+### opam show --raw bad-opam-version.2 | "${OPAMMAJORVERSION}" -> "CURRENTMAJOR"
+opam-version: "2.0"
+name: "bad-opam-version"
+version: "2"
+synopsis: ""
+description: """\
+This package uses opam 5.0 file format which opam CURRENTMAJOR cannot read.
+
+In order to install or view information on this package, please upgrade your opam installation to at least version 5.0."""
+available: opam-version >= "5.0"
+### :: Parse errors
+### <VRS/packages/two-zero/two-zero.1/opam>
+opam-version: "2.0"
+GARBAGE
+### opam update versions
+
+<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
+[versions] synchronised from file://${BASEDIR}/VRS
+[WARNING] Could not read file ${BASEDIR}/OPAM/repo/versions/packages/two-zero/two-zero.1/opam. skipping:
+          At ${BASEDIR}/OPAM/repo/versions/packages/two-zero/two-zero.1/opam:3:0-3:0::
+          Parse error
+### opam show two-zero --raw
+[ERROR] No package matching two-zero found
+# Return code 5 #
+### rm -r VRS/packages
+### <VRS/packages/two-one/two-one.1/opam>
+opam-version: "2.1"
+GARBAGE
+### opam update versions
+
+<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
+[versions] synchronised from file://${BASEDIR}/VRS
+[WARNING] Could not read file ${BASEDIR}/OPAM/repo/versions/packages/two-one/two-one.1/opam. skipping:
+          At ${BASEDIR}/OPAM/repo/versions/packages/two-one/two-one.1/opam:3:0-3:0::
+          Parse error
+Now run 'opam upgrade' to apply any package updates.
+### opam show two-one --raw
+[ERROR] No package matching two-one found
+# Return code 5 #
+### rm -r VRS/packages
+### <VRS/packages/two-two/two-two.1/opam>
+opam-version: "2.2"
```

Comment:
This - and the 2.3 one below - are the ones producing the wrong output (which is tracked in #6188 and which a PR for that will therefore fix)